### DISCUSSION TITLE
Batching Numerous Requests

### DISCUSSION THREAD

**Discussion by kentlhastings** [pnp/pnpcore#373] [Q&A] [answered]
Hello all,

I'm working on a project where we need to update the permissions on a larger number of folders and files, and I was hoping that I could use the batching features of pnp core.

If I start with a list of serverRelativeURLs for each item, is there a way to batch multiple calls to BreakRoleInheritance and AddRoleDefinitionsAsync while still taking advantage of the pnpContext?

As always, thank you for your help!

**Comment by jansenbe** [comment, marked as answer]
@kentlhastings : The current nightly released introduced part of the batch methods you need: https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core/Model/SharePoint/Core/Public/IListItem.cs#L369 is there. I'll add equivalent batch methods for the AddRoleDefinitions method, thanks for raising this.

**Reply by jansenbe** [reply]
See https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core.Test/Security/ListItemSecurityTests.cs#L171 for an example on how to use the new AddRoleDefinitionBatch methods, combined with BreakRoleInheritanceBatch this enables your scenario.

**Reply by kentlhastings** [reply]
Bert, you have been a life saver this week. I really appreciate all of your assistance. My team and I will look forward to the addition of the AddRoleDefinitionBatch. The changes so far have already cut our wait times in half, which is amazing!
